wagtail/admin/edit_handlers.py

```diff
@@ -515,6 +515,10 @@ def get_comparison_class(self):
 
             if isinstance(field, RichTextField):
                 return compare.RichTextFieldComparison
+
+            if isinstance(field, (CharField, TextField)):
```

Comment:
Should we add a check here so that we don't use ``TextFieldComparison`` if ``choices`` are set on ``CharField``?

Reply:
I don't think this check is necessary because we already handle fields with choices [above](https://github.com/wagtail/wagtail/blob/e96ad4433ca584eb469bf3278771e4d7bb6fe0a5/wagtail/admin/edit_handlers.py#L505), using `ChoiceFieldComparison` then instead.